controller/api/destination/watcher/ip_watcher.go

```diff
@@ -389,7 +415,7 @@ func (ss *serviceSubscriptions) unsubscribe(port Port, listener EndpointUpdateLi
 	delete(ss.listeners, listener)
 }
 
-func withPort(pods AddressSet, port Port) AddressSet {
+func WithPort(pods AddressSet, port Port) AddressSet {
```

Comment:
Note that the linter wants this to have a docstring...

Although, I'd suggest moving this to be a method on `AddressSet` so you can do something like `w.PodToAddressSet(pod).WithPort(p)`

Reply:
Good call!